src/terms/local.jl

```diff
@@ -69,21 +69,33 @@ Atomic local potential defined by `model.atoms`.
 """
 struct AtomicLocal end
 function (::AtomicLocal)(basis::PlaneWaveBasis{T}) where {T}
-    model = basis.model
     # pot_fourier is <e_G|V|e_G'> expanded in a basis of e_{G-G'}
     # Since V is a sum of radial functions located at atomic
     # positions, this involves a form factor (`local_potential_fourier`)
     # and a structure factor e^{-i G·r}
+    model = basis.model
+    G_cart = G_vectors_cart(basis)
+
+    form_factors = IdDict{Tuple{Int,T},T}()
+    for G in G_cart
+        q = norm(G)
+        for (igroup, group) in enumerate(model.atom_groups)
+            if !haskey(form_factors, (igroup, q))
+                element = model.atoms[first(group)]
+                form_factors[(igroup, q)] = local_potential_fourier(element, q)
+            end
+        end
+    end
 
-    pot_fourier = map(G_vectors(basis)) do G
-        pot = sum(model.atom_groups) do group
-            element = model.atoms[first(group)]
-            form_factor::T = local_potential_fourier(element, norm(model.recip_lattice * G))
-            form_factor * sum(cis2pi(-dot(G, r)) for r in @view model.positions[group])
+    pot_fourier = map(enumerate(G_vectors(basis))) do (iG, G)
+        q = norm(G_cart[iG])
+        pot = sum(enumerate(model.atom_groups)) do (igroup, group)
+            structure_factor = sum(r -> cis2pi(-dot(G, r)), view(model.positions, group))
```

Comment:
We have never used `view` in the code, but only the `@view` / `@views` macros. We can argue, but I'd keep it with `@view` for consistency.